fuse_core/src/parameter.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2023, Intrinsic.ai
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <fuse_core/parameter.hpp>
+
+#include <cassert>
+#include <string>
+#include <unordered_set>
+
+
+namespace fuse_core
+{
+std::unordered_set<std::string>
+list_parameter_overrides_at_prefix(
+  node_interfaces::NodeInterfaces<node_interfaces::Parameters> interfaces,
+  std::string prefix, size_t depth)
+{
+  const std::map<std::string, rclcpp::ParameterValue> & overrides =
+      interfaces.get_node_parameters_interface()->get_parameter_overrides();
+  return detail::list_parameter_overrides_at_prefix(overrides, prefix, depth);
+}
+
+std::unordered_set<std::string>
+detail::list_parameter_overrides_at_prefix(
+  const std::map<std::string, rclcpp::ParameterValue> & overrides,
+  std::string prefix, size_t depth)
+{
+  // TODO(sloretz) ROS 2 must have this in a header somewhere, right?
```

Comment:
If it exists it probably doesn't exist universally :grimacing: 

Case in point:
https://github.com/ros2/rclcpp/blob/df994e435df6cbdb19ab7ada839f403367fb769a/rclcpp/include/rclcpp/node_impl.hpp#L292